### DISCUSSION TITLE
Switch expression continuation / piping

### DISCUSSION THREAD

**Discussion by BenjaBobs** [dotnet/csharplang#5023] [Language Ideas] [answered]
My apologies if this is a duplicate, I couldn't find a preexisting issue on this.

### I want to be able to call members on the result of a switch expression
However this is not currently allowed syntax:

```csharp
public string CreateBase64Message<TData>(TData data)
{
    // use extension method on the result of the switch expression
    return data switch 
    {
        string stringData => stringData,
        _ => JsonConvert.SerializeObject(data),
    }
    .Base64Encode(); // CS0103 - The name 'Base64Encode' does not exist in the current context
}
```

As far as I can see my options are as follows:

*Wrap in parentheses*
This comes close to the wanted syntax, but the parentheses makes it a bit harder to read.
```csharp
public string CreateBase64Message<TData>(TData data)
{
    return (data switch
    {
        string stringData => stringData.Base64Encode(),
        _ => JsonConvert.SerializeObject(data).Base64Encode(),
    }).Base64Encode();
}
```

*Save a local variable*
With this option the method can no longer be expression bodied.
```csharp
public string CreateBase64Message<TData>(TData data)
{
    var rawMessage = data switch 
    {
        string stringData => stringData,
        _ => JsonConvert.SerializeObject(data),
    };

    return rawMessage.Base64Encode();
}
```

*Call on each branch*
With this option, maintainability is sacrificed.
```csharp
public string CreateBase64Message<TData>(TData data)
{
    return data switch 
    {
        string stringData => stringData.Base64Encode(),
        _ => JsonConvert.SerializeObject(data).Base64Encode(),
    };
}
```

**Comment by Joe4evr** [comment, marked as answer]
This would require changing precedence, which makes for a disastrous IDE experience as explained in #3005.
See also https://github.com/dotnet/csharplang/issues/3234#issuecomment-590992926.

**Reply by BenjaBobs** [reply]
https://github.com/dotnet/csharplang/issues/3234#issuecomment-590992926 does make a lot of sense. Thanks!
